src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -0,0 +1,143 @@
+            CategoryOD categoryODFailAdd = new CategoryOD("Category budget needs to be a number. Please try again!");
+            return categoryOB.fail(categoryODFailAdd);
+
+        } catch (EntityException e) {
+            //EntityException: User tries to add an invalid Category name but failed. (See entities/EntityException.java)
+            CategoryOD categoryODFailAdd = new CategoryOD("There is already a category with this new name in this month.");
+            return categoryOB.fail(categoryODFailAdd);
+        }
+    }
+    /**
+     * Overrides method in CategoryIB.
+     * Attempts to edit a category with information from CategoryID and returns a CategoryOD indicating whether fail/success after execution.
+     * Provides detailed fail messages according to each condition below. (Explained in comments.)
+     * @param categoryIDEdit CategoryID required for editing a new category to the designated monthID MonthlyStorage Object.
+     * @return CategoryOD String message indicating success/fail add attempt.
+     * @throws EntityException (Although we know MonthlyStorage with monthID is always in the SessionStorage,
+     *                          it will be caught at views/add_edit_category_views/EditCategoryV.java).
+     */
+    @Override
+    public CategoryOD editCategoryInMonth(CategoryID categoryIDEdit) throws EntityException{
+        MonthlyStorage month = categoryIDEdit.getSession().getMonthlyData(categoryIDEdit.getMonthID());
+        try {
+            ArrayList<Category> categoryList = month.getCategoryData();
+            double valueDouble = toDouble(categoryIDEdit.getValue());
+            if (valueDouble < 0) {
+                //Category budget less than 0: User tries to edit a budget value with input that is a negative number.
+                CategoryOD categoryODFailEdit = new CategoryOD("Category budget can't be less than $0. Please try again!");
+                return categoryOB.fail(categoryODFailEdit);
+            }
+
+            Category existingCategory = findCategory(categoryList, categoryIDEdit.getOldCategory());
+
+            for (Category category1 : categoryList) {
+                if (category1.getName().equals(categoryIDEdit.getName())) {
+                    //Repeated name: User tries to edit category name to another name that exists in the month.
+                    CategoryOD categoryODFailEdit = new CategoryOD("There is already a category with this new name in this month.");
+                    return categoryOB.fail(categoryODFailEdit);
+                }
+            }
+
+            existingCategory.setName(categoryIDEdit.getName());
+            categoryIDEdit.setValue(valueDouble);
+            existingCategory.setBudget(valueDouble);
+
+            CategoryOD categoryODSuccessEdit = new CategoryOD("You have edited a category!");
+            return categoryOB.success(categoryODSuccessEdit);
```

Comment:
I think you would want this in your try clause.

Reply:
I believe they are!! (catches are below)